contrib/babelfishpg_tds/src/backend/tds/tdsresponse.c

```diff
@@ -500,16 +507,39 @@ resolve_numeric_typmod_from_exp(Node *expr)
 			 * Refer to details of precision and scale calculation in the following link:
 			 * https://github.com/MicrosoftDocs/sql-docs/blob/live/docs/t-sql/data-types/precision-scale-and-length-transact-sql.md
 			 */
+			has_aggregate_operand = arg1->type == T_Aggref ||
+						(list_length(op->args) == 2 && arg2->type == T_Aggref);
+
 			switch (op->opfuncid)
 			{
 				case NUMERIC_ADD_OID:
 				case NUMERIC_SUB_OID:
+					integralDigitCount = Max(precision1 - scale1, precision2 - scale2);
 					scale = Max(scale1, scale2);
-					precision = Max(precision1 - scale1, precision2 - scale2) + 1 + scale;
+					precision = integralDigitCount + 1 + scale;
+					/*
+					 * For addition and subtraction, skip scale adjustment when none of the
+					 * operands is part of any aggregate function
+					 */
+					if (has_aggregate_operand &&
+					    integralDigitCount < (Min(TDS_MAX_NUM_PRECISION, precision) - scale))
+						scale = Min(precision, TDS_MAX_NUM_PRECISION) - integralDigitCount;
```

Comment:
this will make scale = scale + 1, why do we want to do that? scale = (integralDigitCount + 1 + scale) - integralDigitCount

Reply:
To match SQL server behaviour, these rules can be found here: https://github.com/MicrosoftDocs/sql-docs/blob/live/docs/t-sql/data-types/precision-scale-and-length-transact-sql.md